### DISCUSSION TITLE
Which version do I need for ESP32-PICO-KIT?

### DISCUSSION THREAD

**Discussion by ygoe** [micropython/micropython#9711] [Hardware & Peripherals]
My development board is an Espressif ESP32-PICO-KIT V4.1. I'd like to try out MicroPython but don't know which version to download. There's three versions:

* D0WD – Should be good for ESP32-WROOM-32x modules which have 4 MB external SPI flash (must be hidden under the metal shield)
* D2WD – Has integrated 2 MB flash
* D0WD with OTA support – What does the OTA refer to here, MicroPython runtime or the user .py scripts?

My kit seems to have a different chip which is called ESP32-PICO-D4. There is no visible flash chip on the board, just the ESP32 chip (black plastic, QFN size, not a metal shield). Can I use MicroPython with it?

**Comment by davefes** [comment]
A suggestion (on Linux) is using esptool do a:
```py
esptool.py --chip esp32 --port /dev/ttyUSB0 erase_flash
```
and that will tell you which chip you have.

**Reply by robert-hh** [reply]
The chip is a Pico D4 with 4 MB flash. You can use the GENERIC firmware with it. It may be helpful to perform a flash erase as first step, as mentioned above,

**Reply by TomLinz** [reply]
How about the successor board ESP32-PICO-KIT-1, which MicroPython version is suitable for it? Instead of a PICO-D4 chip, it is based on PICO-V3.
